redis/src/main/scala/akka/streams/alpakka/redis/scaladsl/RedisFlow.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.streams.alpakka.redis.scaladsl
+
+import akka.NotUsed
+import akka.stream.scaladsl.Flow
+import akka.streams.alpakka.redis
+import akka.streams.alpakka.redis.{RedisKeyValue, RedisKeyValues, RedisOperationResult, RedisPubSub}
+import io.lettuce.core.api.StatefulRedisConnection
+import io.lettuce.core.pubsub.StatefulRedisPubSubConnection
+
+import scala.collection.JavaConverters._
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+object RedisFlow {
+
+  def set[K, V](parallelism: Int, connection: StatefulRedisConnection[K, V])(
+      implicit executionContext: ExecutionContext
+  ): Flow[RedisKeyValue[K, V], RedisOperationResult[RedisKeyValue[K, V], String], NotUsed] =
```

Comment:
>  think parallelism and connection should be part of a settings object. This way, the API can evolve without breaking.

Yes. This is good point. Will do that. 

> I've seen that a couple other connectors require executionContext on flow creation - this is something I'd ask @ennru or @2m about their opinion on. I would rather have the stage use Akka Stream's dispatchers. How that would work is something you can see from ElasticSearch / Sqs connectors (using a stage producing Futures followed by mapAsync(parallelism)(identity)) For how to get the execution context in a stage, have a look at the JMS flow.

For getting Default Akka dispatcher I need to inherit GraphStageLogic and I don't see any reason for that except using Akka-Streams default dispatchers. I think it might be useful to use dedicated dispatcher for specific external IO operation and monitor it using Kamon.io or Cinnamon.   

> I'm also not sure about passing the connection into the flow operation vs. passing in the Redis client and creating a connection within the stage (and closing it once the stage completes). Can you elaborate why you chose the first option?

Doing that will be extremely expensive in terms of performance especially at high throughput in stage. Sometimes it might be cheaper to keep connection open than reopen it every time. Evolution of HTTP protocol from 1.0 to 1.1 and then to HTTP 2.0 is good example for that.

Reply:
Sorry for being unclear in my comment on connections. I wouldn't recommend creating a new connection for each Redis operation - that would indeed reduce the max achievable throughput drastically. What I'm thinking about is lifting the burden of connection management from the user, and manage the connection in the Alpakka Redis connector. Looking at other connectors however (more specifically, the ElasticSearch one), it looks like passing the Redis connection into the stage and let the user take care of closing the connection is a pattern that others follow as well.